k9mail/src/main/java/com/fsck/k9/account/AccountCreator.java

```diff
@@ -0,0 +1,31 @@
+package com.fsck.k9.account;
+
+import com.fsck.k9.Account.DeletePolicy;
+import com.fsck.k9.mail.ServerSettings.Type;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Deals with logic surrounding account creation.
+ * <p/>
+ * TODO Move much of the code from com.fsck.k9.activity.setup.* into here
+ */
+public class AccountCreator {
+
+    private static Map<Type, DeletePolicy> defaults;
+
+    public static DeletePolicy calculateDefaultDeletePolicy(Type type) {
+        return getDefaults().get(type);
+    }
+
+    private static synchronized Map<Type, DeletePolicy> getDefaults() {
+        if (null == defaults) {
```

Comment:
Please follow the K-9 code style. We always put `null` on the right side of a comparison.

You could avoid `synchronized` on this method by putting the initialization code in a static initializer.